src/components/Whiteboard/ElementBehaviors/Resizable/utils.ts

```diff
@@ -80,6 +81,89 @@ export function calculateDragDimension(
   };
 }
 
+export function calculateLineDragDimensions(
+  handlePosition: ResizeHandlePosition,
+  dimensions: Dimensions,
+  dragX: number,
+  dragY: number,
+): Dimensions {
+  if (
+    dimensions.elementKind !== 'line' ||
+    handlePosition === 'top' ||
+    handlePosition === 'right' ||
+    handlePosition === 'bottom' ||
+    handlePosition === 'left'
+  ) {
+    return dimensions;
+  }
+
+  const start = dimensions.points[0];
+  const end = dimensions.points[dimensions.points.length - 1];
+  const cursor = { x: dragX - dimensions.x, y: dragY - dimensions.y };
+
+  let newPoints;
+
+  switch (handlePosition) {
+    case 'topRight':
```

Comment:
In my opinion lines should only have two handles: start and end.
Then this entire code here would be a no-brainer.

I wonder if things are getting too complicated here.
Resize of paths is already covered by polyline resize. What should other use-cases of resizing paths be? Also would then only one point change it's position?

Happy to discuss with you or the others.

Reply:
Fine by me. Just pulled out my stashed solution which I held back for certain reasons 🙂

Reply:
Thanks, much better :slightly_smiling_face:  Does the same with less code.